openedx/core/djangoapps/content_libraries/docs/decisions/0004-upstream-block.rst

```diff
@@ -0,0 +1,89 @@
+4. Persist the relationship between blocks and their upstream source
+####################################################################
+
+Status
+******
+
+Accepted
+
+Pending implementation in (TODO - link)
+
+Context
+*******
+
+* TODO - context
+
+Decision
+********
+
+* The following two fields will be added to every XBlock in CMS:
+
+  .. code-block:: python
+
+     upstream_block = String(
+         scope=Scope.settings,
+         help=(
+             "The usage key of a the block that served as this block's template, if one exists. "
```

Comment:
typo: `of a the block`

But also, is "template" the right word? To me that generally implies a one-time copy-and-then-modify approach, but using library content is generally more clone-and-keep-in-sync.

Reply:
Good point. How does this look @bradenmacdonald ? I also added a note about nonexistent upstream blocks.
```
The usage key of a block (generally within a Content Library) which serves as a source of upstream
updates for this block, or None if there is no such upstream. Please note: It is valid for upstream_block
to hold a usage key for a block that does not exist (or does not *yet* exist) on this instance,
particularly if this block was imported from a different instance.
```

Reply:
I like that, @kdmccormick 👍🏻